test/utils/keys.test.ts

```diff
@@ -65,10 +65,27 @@ describe('AttributesUtil', () => {
       const filtered = Keys.filter(payload, 'preferences.email.news');
       expect(filtered).to.deep.equal({ preferences: { email: { news: true } } });
     });
+    it('should keep all nested properties', () => {
+      const filtered = Keys.filter(payload, 'preferences.*');
+      expect(filtered).to.deep.equal({
+        preferences: {
+          email: { news: true, newPost: false },
+          sms: { newComment: true }
+        }
+      });
+    });
     it('should keep nested array property', () => {
       const filtered = Keys.filter(payload, 'posts.[].title');
       expect(filtered).to.deep.equal({ posts: [{ title: 'RBAC' }, { title: 'ABAC' }] });
     });
+    it('should keep all nested array property', () => {
+      const filtered = Keys.filter(payload, '*.[].title');
+      expect(filtered).to.deep.equal({ posts: [{ title: 'RBAC' }, { title: 'ABAC' }] });
+    });
+    it('should keep all nested array property', () => {
+      const filtered = Keys.filter(payload, '*.[].content');
+      expect(filtered).to.deep.equal({ 'posts': [{ 'comments': [{ 'content': 'Super cool' }, { 'content': 'Ugh?' }, {}], 'content': 'Lorem ipsum' }, { 'content': 'Lorem ipsum' }] });
+    });
     it('should keep full array', () => {
```

Comment:
Try:
```
const filtered = Keys.filter(payload, '*tle');
expect(filtered).to.deep.equal({ posts: [{ title: 'RBAC' }, { title: 'ABAC' }] });
```

Weird, isn't it?

Reply:
Hm yeah good catch. There would need to be more protection. A `*` would need to be surrounded by `.`s.